src/Sylius/Behat/Page/Shop/Account/AddressBook/IndexPage.php

```diff
@@ -41,97 +35,72 @@ public function getAddressesCount()
         return $addressesCount;
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function hasAddressOf($fullName)
+    public function hasAddressOf(string $fullName): bool
     {
         return null !== $this->getAddressOf($fullName);
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function hasNoAddresses()
+    public function hasNoAddresses(): bool
     {
-        return $this->getDocument()->hasContent('You have no addresses defined');
+        return $this->hasElement('messages');
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function addressOfContains($fullName, $value)
+    public function addressOfContains(string $fullName, string $value): bool
     {
-        $address = $this->getAddressOf($fullName);
+        $address = $this->getAddressPageOf($fullName);
 
-        return $address->has('css', sprintf('address:contains("%s")', $value));
+        return $address->has('css', sprintf('[data-test-address]:contains("%s")', $value));
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function editAddress($fullName)
+    public function editAddress(string $fullName): void
     {
-        $addressToEdit = $this->getAddressOf($fullName);
-        $addressToEdit->findLink('Edit')->press();
+        $addressToEdit = $this->getAddressPageOf($fullName);
+        $addressToEdit->find('css', '[data-test-edit-button] [data-test-button]')->press();
```

Comment:
IMO you should add information what the second button is that, and do we need both attributes?